assets/components/kube-controller-manager/defaultconfig.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: kubecontrolplane.config.openshift.io/v1
+kind: KubeControllerManagerConfig
+extendedArguments:
+  enable-dynamic-provisioning:
+  - "true"
+  allocate-node-cidrs:
+  - "false"
+  configure-cloud-routes:
+  - "false"
+  use-service-account-credentials:
+  - "true"
+  flex-volume-plugin-dir:
+  - "/etc/kubernetes/kubelet-plugins/volume/exec" # created by machine-config-operator, owned by storage team/hekumar@redhat.com
+  pv-recycler-pod-template-filepath-nfs: # owned by storage team/fbertina@redhat.com
+  - "/etc/kubernetes/static-pod-resources/configmaps/recycler-config/recycler-pod.yaml"
+  pv-recycler-pod-template-filepath-hostpath: # owned by storage team/fbertina@redhat.com
+  - "/etc/kubernetes/static-pod-resources/configmaps/recycler-config/recycler-pod.yaml"
```

Comment:
@benluddy  i think we should remove the above keys, up to `flex-volume-plugin-dir`

Reply:
I would agree.

Flex volumes plugins are deprecated in both k8s and OCP in favor of CSI drivers:
- https://kubernetes.io/docs/concepts/storage/volumes/#flexvolume
- https://docs.openshift.com/container-platform/4.11/storage/persistent_storage/persistent-storage-flexvolume.html

Same for custom recyclers: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#recycle, they are replaced by dynamic provisioning.

Reply:
The pattern we've established is to have the [scripts/auto-rebase/rebase.sh](https://github.com/openshift/microshift/blob/main/scripts/auto-rebase/rebase.sh#L395) script copy the asset from the upstream component and use yq to modify it in place with microshift-specific changes. That way the file checked into git here reflects what we want at runtime and we have automation to keep those changes in place each time we update the dependencies.